src/test/java/org/wildfly/security/ldap/DirContextFactoryRule.java

```diff
@@ -47,13 +64,129 @@
+                .setSigningKey(issuerSelfSignedX509CertificateAndSigningKey.getSigningKey())
+                .setPublicKey(scarabPublicKey)
+                .setSerialNumber(new BigInteger("4"))
+                .addExtension(new BasicConstraintsExtension(false, false, -1))
+                .build();
+        scarabKeyStore.setKeyEntry("scarab", scarabSigningKey, PASSWORD, new X509Certificate[]{scarabCertificate,issuerCertificate});
+
+        // Modify the ldif
+        MessageDigest md = MessageDigest.getInstance("SHA-1");
+        String digest = ByteIterator.ofBytes(md.digest(scarabCertificate.getEncoded())).hexEncode(true).drainToString();
+
+        File workingDirLDIF = new File(LDAP_DIRECTORY_LOCATION);
+        if (workingDirLDIF.exists() == false) {
+            workingDirLDIF.mkdirs();
+        }
+
+        CodePointIterator certificateBytes = ByteIterator.ofBytes(scarabCertificate.getEncoded()).base64Encode();
+        String certificateBaseString = "usercertificate:: " + certificateBytes.drainToString();
+        String certificateString = "";
+        int counter = 0;
+        for (int i = 0; i < certificateBaseString.length(); i++){
+            if(i == 78 || i == (78+77*counter)){
+                certificateString = certificateString + System.getProperty("line.separator");
+                certificateString = certificateString + " ";
+                counter += 1;
+            }
+            certificateString = certificateString + certificateBaseString.charAt(i);
+        }
+
+        FileWriter ldif = new FileWriter(workingDirLDIF.toString() + LDIF_LOCATION, true);
+        ldif.write("x509digest: " + digest + System.getProperty("line.separator"));
+        ldif.write(certificateString);
+        ldif.close();
+    }
+
+    private static void setUp() throws Exception{
+        File workingDirCA = new File(CA_JKS_LOCATION);
+        if (workingDirCA.exists() == false) {
+            workingDirCA.mkdirs();
+        }
+
+        KeyStore localhostKeyStore = KeyStore.getInstance("JKS");
+        localhostKeyStore.load(null, null);
+        KeyStore scarabKeyStore = KeyStore.getInstance("JKS");
+        scarabKeyStore.load(null, null);
+        KeyStore trustStore = KeyStore.getInstance("JKS");
+        trustStore.load(null, null);
+
+        File localhostFile = new File(workingDirCA, "localhost.keystore");
+        File scarabFile = new File(workingDirCA, "scarab.keystore");
+        File trustFile = new File(workingDirCA, "ca.truststore");
+
+        createStores(localhostKeyStore, scarabKeyStore, trustStore);
+        createStoreFiles(localhostFile, localhostKeyStore, scarabFile, scarabKeyStore, trustFile, trustStore);
+    }
 
     @Override
     public Statement apply(Statement current, Description description) {
         return new Statement() {
 
             @Override
             public void evaluate() throws Throwable {
+                try {
+                    setUp();
+                } catch (Exception e) {}
```

Comment:
If ```setUp()``` throws an exception for some reason, the empty catch block here will prevent it from being reported. Since ```evaluate()``` is already declared as throwing ```Throwable```, the try/catch block could just be removed.

Reply:
Fixed